backend/sozisel/lib/sozisel_web/schema/resolvers/participant_resolvers.ex

```diff
@@ -186,4 +191,90 @@ defmodule SoziselWeb.Schema.Resolvers.ParticipantResolvers do
 
     verify_launched_event(ctx, launched_event_id, on_verify)
   end
+
+  def insert_result_in_transaction(event_result_attrs, image, filename, extension) do
+    Ecto.Multi.new()
+    |> Ecto.Multi.insert(
+      :whiteboard_event_result,
+      EventResult.create_changeset(%EventResult{}, event_result_attrs)
+    )
+    |> Ecto.Multi.run(:file_rename, fn _repo, event_result ->
+      %{
+        whiteboard_event_result: %{
+          result_data: %{path: path}
+        }
+      } = event_result
+
+      processed_image_path = Path.rootname(path) <> "_processed" <> extension
+
+      with :ok <- File.touch(processed_image_path),
+           :ok <- File.rename(image.path, processed_image_path),
+           :ok <- @media_storage_module.store_file(filename, processed_image_path) do
+        {:ok, %{}}
+      else
```

Comment:
You don't need the else statement (it will return whatever has not been matched) unless you add some logging. And I think it is a proper place to match on error tuple and log the error.